app/src/main/res/layout/fragment_data.xml

```diff
@@ -1,132 +1,248 @@
+        app:cardCornerRadius="20dp"
+        app:cardElevation="10dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/fragment_data_title">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:id="@+id/tree_data_layout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:paddingLeft="@dimen/sixteen_dp"
+            android:paddingTop="@dimen/padding_five"
+            android:paddingRight="@dimen/sixteen_dp"
+            android:paddingBottom="@dimen/padding_five">
+
+            <androidx.constraintlayout.widget.Guideline
+                android:id="@+id/tree_data_guideline"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:orientation="vertical"
+                app:layout_constraintGuide_begin="243dp" />
+
+            <TextView
+                android:id="@+id/fragmentDataUploadedLabel"
+                android:layout_width="0dp"
+                android:layout_height="@dimen/height_forty"
+                android:layout_marginStart="5dp"
+                android:layout_marginTop="16dp"
+                android:layout_weight="3"
+                android:text="@string/uploaded"
+                android:textAppearance="@style/TextAppearance.AppCompat.Medium"
+                android:textColor="@color/black"
+                android:textSize="@dimen/text_size_seventeen"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/fragmentDataUploaded"
+                android:layout_width="0dp"
+                android:layout_height="@dimen/height_forty"
+                android:layout_marginTop="16dp"
+                android:layout_weight="3"
+                android:textColor="@color/black"
+                android:textIsSelectable="true"
+                android:textSize="@dimen/text_size_seventeen"
+                app:layout_constraintStart_toStartOf="@+id/tree_data_guideline"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/fragmentDataWaitingUploadLabel"
+                android:layout_width="0dp"
+                android:layout_height="@dimen/height_forty"
+                android:layout_marginTop="16dp"
+                android:layout_weight="3"
+                android:text="@string/fragment_data_waiting_to_upload_label"
+                android:textColor="@color/black"
+                android:textSize="@dimen/text_size_seventeen"
+                app:layout_constraintStart_toStartOf="@+id/fragmentDataUploadedLabel"
+                app:layout_constraintTop_toBottomOf="@+id/fragmentDataUploadedLabel" />
+
+            <TextView
```

Comment:
For all the textviews that need to be populated by data, can we please add debug text to it?
e.g. `tools:text="$12.45"`

Reply:
Adding them.